### PR TITLE
refactor(selector): move Dockerfile() and Schedule() into selector package

### PR DESCRIPTION
**Pull request by bvtujo**
Moves the prompting logic for Dockerfiles and job schedules into the selector package so the CLI package doesn't have to handle it. 

related #1390 and #1131 : Prerequisite to job init being complete. 

Work in progress

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/interfaces.go [hunk 1]
```diff
@@ -387,6 +393,15 @@ type wsSelector interface {
 	Job(prompt, help string) (string, error)
 }
 
+type initJobSelector interface {
+	Dockerfile(selPrompt, getPrompt, selHelp, getHelp string, pv prompt.ValidatorFunc) (string, error)
+	Schedule(scheduleTypePrompt, scheduleTypeHelp string, scheduleValidator, rateValidator prompt.ValidatorFunc) (string, error)
+}
+
+type initSvcSelector interface {
```
Comment (efekarakus) [inline internal/pkg/cli/interfaces.go]: nit: `dockerfileSelector`

File: internal/pkg/cli/interfaces.go [hunk 2]
```diff
@@ -387,6 +393,15 @@ type wsSelector interface {
 	Job(prompt, help string) (string, error)
 }
 
+type initJobSelector interface {
+	Dockerfile(selPrompt, getPrompt, selHelp, getHelp string, pv prompt.ValidatorFunc) (string, error)
+	Schedule(scheduleTypePrompt, scheduleTypeHelp string, scheduleValidator, rateValidator prompt.ValidatorFunc) (string, error)
+}
```
Comment (efekarakus) [inline internal/pkg/cli/interfaces.go]:
```suggestion
type initJobSelector interface {
	dockerfileSelector
	Schedule(scheduleTypePrompt, scheduleTypeHelp string, scheduleValidator, rateValidator prompt.ValidatorFunc) (string, error)
}
```

File: internal/pkg/cli/job_init.go [hunk 3]
```diff
@@ -29,55 +28,17 @@ var (
...
-	fixedSchedule = "Fixed Schedule"
-
-	custom  = "Custom"
-	hourly  = "Hourly"
-	daily   = "Daily"
-	weekly  = "Weekly"
-	monthly = "Monthly"
-	yearly  = "Yearly"
-
+// const (
```
Comment (efekarakus) [inline internal/pkg/cli/job_init.go]: can we remove these comments?

File: internal/pkg/term/selector/selector.go [hunk 4]
```diff
@@ -44,6 +93,12 @@ type WsWorkloadLister interface {
 	JobNames() ([]string, error)
 }
 
+// WsConfigGetter wraps methods to get workload names and Dockerfiles from the workspace.
+type WsConfigGetter interface {
```
Comment (efekarakus) [inline internal/pkg/term/selector/selector.go]: nit:
```suggestion
// WsWorkloadDockerfileLister wraps methods to get workload names and Dockerfiles from the workspace.
type WsConfigGetter interface {
```

File: internal/pkg/term/selector/selector.go [hunk 5]
```diff
@@ -373,3 +428,133 @@ func (s *WorkspaceSelect) retrieveWorkspaceJobs() ([]string, error) {
 	}
 	return localJobNames, nil
 }
+
+// Dockerfile asks the user to select from a list of Dockerfiles in the current
+// directory or one level down. If no dockerfiles are found, it asks for a custom path.
+func (s *WorkspaceSelect) Dockerfile(selPrompt, getPrompt, selHelp, getHelp string, pathValidator prompt.ValidatorFunc) (string, error) {
```
Comment (efekarakus) [inline internal/pkg/term/selector/selector.go]: nit:
```suggestion
func (s *WorkspaceSelect) Dockerfile(selPrompt, notFoundPrompt, selHelp, notFoundHelp string, pathValidator prompt.ValidatorFunc) (string, error) {
```

File: internal/pkg/workspace/workspace.go [hunk 6]
```diff
@@ -397,3 +400,61 @@ func (ws *Workspace) read(elem ...string) ([]byte, error) {
 	pathElems := append([]string{copilotPath}, elem...)
 	return ws.fsUtils.ReadFile(filepath.Join(pathElems...))
 }
+
+// ListDockerfiles returns the list of Dockerfiles within the current
+// working directory and a sub-directory level below. If an error occurs while
+// reading directories, or no Dockerfiles found returns the error.
+func (ws *Workspace) ListDockerfiles() ([]string, error) {
```
Comment (efekarakus) [inline internal/pkg/workspace/workspace.go:407]: nice!

**Review by efekarakus — APPROVED** [review]
lgtm! some small renames

**Review by iamhopaul123 — APPROVED** [review]
ship it :shipit:

**Pull request merged**